batch.go

```diff
@@ -124,6 +124,18 @@ func (t *Batch) Add(nd Node) error {
 	return t.err
 }
 
+// Add many calls Add for every given Node, thus batching and
+// commiting them as needed.
+func (t *Batch) AddMany(ctx context.Context, nodes []Node) error {
+	for _, n := range nodes {
+		err := t.Add(ctx, n)
```

Comment:
Could potentially append all nodes directly but that might get us way over maxSize/maxNodes.

Reply:
That shouldn't really be an issue. Going over max nodes/size should just trigger a flush (we don't want to buffer too much in memory).

Maybe we should just:

1. Make `Add` call AddMany`.
2. Move the main logic to `AddMany` (and process all nodes at once).